rfcs/014-pagination.md

```diff
@@ -0,0 +1,193 @@
+---
+name: Pagination support
+about: Add bookmark based pagination interface
+title: 'Pagination API'
+labels: rfc, discussion
+assignees: ''
+
+---
+
+Implement pagination support for query like endpoints
+
+# Introduction
+
+The main driver for this proposal is the introduction of FoundationDB as a
+storage engine. FoundationDB imposes limits on transaction duration and size.
+Therefore we need to find way to restrict amount of data we return to customers.
```

Comment:
FDB does have limits on transactions however those don't necessarily translate to limits on request sizes and API shape as discussed in ML. For instance, in the current version prototype/fdb-layer branch the API will behave in a compatible manner to 2.x and 3.x and stream all the data to the users, even if it is 1-2M docs/rows.

I think we could say it is beneficial to have reasonable limits to encourage a better application design and help avoid operational issues with accidentally streaming all the data in one request. But CouchDB users who already rely on this, maybe not ideal, behavior should be able to port their application to 4.x and then slowly migrate or improve their design.